test/e2e/operator.go

```diff
@@ -311,22 +311,54 @@ var _ = Describe("ARO Operator - Azure Subnet Reconciler", func() {
 })
 
 var _ = Describe("ARO Operator - MUO Deployment", func() {
+	const (
+		managedUpgradeOperatorNamespace  = "openshift-managed-upgrade-operator"
+		managedUpgradeOperatorDeployment = "managed-upgrade-operator"
+	)
+
+	deleteMUODeployment := func(ctx context.Context) error {
+		return clients.Kubernetes.
+			AppsV1().
+			Deployments(managedUpgradeOperatorNamespace).
+			Delete(ctx, managedUpgradeOperatorDeployment, metav1.DeleteOptions{})
+	}
+
+	muoDeploymentExists := func(g Gomega, ctx context.Context) {
+		_, err := clients.Kubernetes.
+			AppsV1().
+			Deployments(managedUpgradeOperatorNamespace).
+			Get(ctx, managedUpgradeOperatorDeployment, metav1.GetOptions{})
+
+		g.Expect(err).ToNot(HaveOccurred())
+	}
+
 	It("must be deployed by default with FIPS crypto mandated", func(ctx context.Context) {
 		By("getting MUO pods")
-		pods, err := clients.Kubernetes.CoreV1().Pods("openshift-managed-upgrade-operator").List(ctx, metav1.ListOptions{
+		pods, err := clients.Kubernetes.CoreV1().Pods(managedUpgradeOperatorNamespace).List(ctx, metav1.ListOptions{
 			LabelSelector: "name=managed-upgrade-operator",
 		})
 		Expect(err).NotTo(HaveOccurred())
 		Expect(pods.Items).NotTo(BeEmpty())
 
 		By("verifying that MUO has FIPS crypto mandated by reading logs")
 		Eventually(func(g Gomega, ctx context.Context) {
-			b, err := clients.Kubernetes.CoreV1().Pods("openshift-managed-upgrade-operator").GetLogs(pods.Items[0].Name, &corev1.PodLogOptions{}).DoRaw(ctx)
+			b, err := clients.Kubernetes.CoreV1().Pods(managedUpgradeOperatorNamespace).GetLogs(pods.Items[0].Name, &corev1.PodLogOptions{}).DoRaw(ctx)
 			g.Expect(err).NotTo(HaveOccurred())
 
 			g.Expect(string(b)).To(ContainSubstring(`msg="FIPS crypto mandated: true"`))
 		}).WithContext(ctx).Should(Succeed())
 	})
+
+	It("should reconcile the MUO deployment when it is delated", func(ctx context.Context) {
```

Comment:
You should be able to read it as a sentence: "ARO Operator - MUO Deployment must be restored if delated" vs "ARO Operator - MUO Deployment should reconcile the MUO deployment when it is delated".

Another nit: use "must". "Should" sounds indecisive in this context :) 

```suggestion
	It("must be restored if delated", func(ctx context.Context) {
```

Reply:
Done. Thanks!